crates/editor/src/highlight_matching_bracket.rs

```diff
@@ -13,7 +14,11 @@ pub fn refresh_matching_bracket_highlights(editor: &mut Editor, cx: &mut ViewCon
         return;
     }
 
-    let head = newest_selection.head();
+    let mut head = newest_selection.head();
+    // Apply 1 offset for vim normal mode
+    if editor.cursor_shape == CursorShape::Block || editor.cursor_shape == CursorShape::Hollow {
+        head += 1;
+    }
```

Comment:
maybe this should be function on selection that take cursor shape as argument

same thing happen in symbol highlighting